### PR TITLE
tensors, distance metric, partial update

### PR DESCRIPTION
**Pull request by kkraune**
@geirst

### CODE REVIEW

File: en/attributes.html [hunk 1]
```diff
@@ -123,10 +141,19 @@ <h2 id="data-structures">Data structures</h2>
       One large memory buffer holding references (32 bit) to a memory structure
       with objects per unique value.
     </td>
+  </tr><tr>
+    <td>Tensor</td>
+    <td>Fixed / Variable</td>
+    <td>
+      Like the "D" attribute - a tensor is a multivalued datatype with sparse and/or dense dimensions,
```
Comment (geirst) [inline en/attributes.html:148]: We should use the terms "mapped and indexed dimensions" to be aligned with https://docs.vespa.ai/en/reference/tensor.html. I am also not sure we should say its a multivalue datatype.

File: en/attributes.html [hunk 2]
```diff
@@ -123,10 +141,19 @@ <h2 id="data-structures">Data structures</h2>
...
+      stored in the <em>tensor store</em>.
+      Dense dimensions resemble the attribute multivalue mapping, as elements are fixed in size and numbers.
```
Comment (geirst) [inline en/attributes.html:150]: I would skip the comparison with the other data stores, as I find it a bit misleading. Consider the following simplification instead: "The memory layout in the tensor store depends on the tensor type".

**Review by geirst — APPROVED** [review]
👍

File: en/attributes.html (hunk 1 above)
Comment (kkraune) [reply en/attributes.html:148]: "mapped and indexed dimensions" better, yes. removed multivalue

Comment (kkraune) [reply en/attributes.html:148]: not sure what we put in multivalue semantics, though, as a tensor has multiple values, but nevermind

File: en/attributes.html (hunk 2 above)
Comment (kkraune) [reply en/attributes.html:150]: ok!

**Review by geirst — COMMENTED** [review]
👍

**Pull request merged**